engine/runtime/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -378,17 +378,22 @@ public void onReturnValue(EventContext context, VirtualFrame frame, Object resul
         }
         callsCache.setExecuted(nodeId);
       } else if (node instanceof ExpressionNode) {
+        boolean isPanic = result instanceof PanicSentinel;
         UUID nodeId = ((ExpressionNode) node).getId();
         String resultType = Types.getName(result);
-        cache.offer(nodeId, result);
+
+        if (!isPanic) {
+          cache.offer(nodeId, result);
+        }
```

Comment:
Can you put a source note here that explains why this is necessary and the bug that it prevents?